### PR TITLE
Merge develop to master for release 4.0.0

### PR DESCRIPTION
**Pull request by zhangcc01**
## Description

Merge develop to master for release 4.0.0.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+
+### App Center
+
+#### Android
+
+* **[Breaking change]** Bumping the minimum Android SDK version to 21 API level (Android 5.0), because old Android versions do not support root certificate authority used by App Center and would not get CA certificates updates anymore.
+
+### iOS
+
+* **[Feature]** **[Breaking change]** Make SDK API more Swift-friendly. You can find the API updates in the [swift migration table](https://docs.microsoft.com/appcenter/sdk/getting-started/migration/apple-sdk-update#swift-api-changes).
```
Comment (MatkovIvan) [inline CHANGELOG.md:13]: Not related

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+* **[Breaking change]** The App Center prefix `MS` in Objective-C is consolidated to `MSAC` for collision prevention. Please follow the [migration guide](https://docs.microsoft.com/appcenter/sdk/getting-started/migration/apple-sdk-update) for adopting the new naming.
```
Comment (MatkovIvan) [inline CHANGELOG.md:14]: No need to migration, but we need to mention that we fixed conflicts

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+* **[Fix]** Fix `Undefined symbols for architecture x86_64` on Xcode 12.
+* **[Fix]** Fix `NSInvalidArgumentException` when using non-string object as a key in `NSUserDefaults`.
+* **[Fix]** Fix `NSDateFormatter` initialization in a concurrent environment.
+
+### App Center Crashes
+
+#### Android
+
+* **[Feature]** Convert the `saveUncaughtException` method to public and return error log identifier when calling it.
```
Comment (MatkovIvan) [inline CHANGELOG.md:23]: Not related

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: AppCenterReactNativeShared/ios/Podfile [hunk 4]
```diff
@@ -1,8 +1,10 @@
+source 'https://msmobilecenter.visualstudio.com/SDK/_git/AppCenterSDK-Specs-Private'
```
Comment (MatkovIvan) [inline AppCenterReactNativeShared/ios/Podfile:1]: Why do we need this change?

**Review by russelarms — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**